SignalServiceKit/src/Storage/OWSSessionStorage.m

```diff
@@ -0,0 +1,161 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSSessionStorage.h"
+#import "OWSFileSystem.h"
+#import "OWSStorage+Subclass.h"
+#import "TSDatabaseView.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+NSString *const OWSSessionStorageExceptionName_CouldNotCreateDatabaseDirectory
+    = @"OWSSessionStorageExceptionName_CouldNotCreateDatabaseDirectory";
+
+#pragma mark -
+
+@interface OWSSessionStorage ()
```

Comment:
This is just a very rough first draft - this class gets considerably refined in subsequent PRs.